### PR TITLE
Add Support for Azure Cosmos DB MongoDB vCore Vector Store #11627

### LINKED ISSUE #11627: Add AzureCosmosDBVectorSearch VectorStore
### Feature request

Azure Cosmos DB for MongoDB vCore enables users to efficiently store, index, and query high dimensional vector data stored directly in Azure Cosmos DB for MongoDB vCore. It contains similarity measures such as COS (cosine distance), L2 (Euclidean distance) or IP (inner product) which measures the distance between the data vectors and your query vector. The data vectors that are closest to your query vector are the ones that are found to be most similar semantically and retrieved during query time. The accompanying PR would add support for Langchain Python users to store vectors from document embeddings generated from APIs such as Azure OpenAI Embeddings or Hugging Face on Azure.

[Azure Cosmos DB for MongoDB vCore](https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/vcore/vector-search)

### Motivation

This capability described in the feature request is currently not available for Langchain Python.

### Your contribution

I will be submitting a PR for this feature request.

### PR DESCRIPTION
**Pull request by izzymsft**
This PR adds support for the Azure Cosmos DB MongoDB vCore Vector Store

https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/vcore/
https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/vcore/vector-search

Summary:
  - **Description:** added vector store integration for  Azure Cosmos DB MongoDB vCore Vector Store, 
  - **Issue:** the issue # it fixes #11627,
  - **Dependencies:** pymongo dependency,
  - **Tag maintainer:** @hwchase17,
  - **Twitter handle:** @izzyacademy

### CODE REVIEW

**Comment by izzymsft** [comment]
@jcodella when you have a moment, please take a look and share some feedback. Thanks.

**Review by izzymsft — COMMENTED** [review]
Thanks @baskaryan for the edits

File: docs/docs/integrations/providers/azure_cosmos_db.mdx
[the remark this answers is not in the document]
Comment (izzymsft) [reply docs/docs/integrations/providers/azure_cosmos_db.mdx:1]: @baskaryan the changes have been made, committed and pushed to the branch. When you have a moment, please take a look and let me know if there are other changes necessary. Thanks.

**Comment by baskaryan** [comment]
lgtm, thanks @izzymsft!!

**Pull request merged**